ui/pages/confirmations/components/confirm/network-change-toast/network-change-toast-inner.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { Box } from '../../../../../components/component-library';
+import { Toast } from '../../../../../components/multichain';
+import {
+  getLastInteractedConfirmationInfo,
+  setLastInteractedConfirmationInfo,
+} from '../../../../../store/actions';
+import { getCurrentChainId } from '../../../../../selectors';
+import { NETWORK_TO_NAME_MAP } from '../../../../../../shared/constants/network';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+
+const MILLISECONDS_IN_ONE_MINUTE = 60000;
+const MILLISECONDS_IN_FIVE_SECONDS = 5000;
+
+const NetworkChangeToastInner = ({
+  confirmation,
+}: {
+  confirmation: { id: string };
+}) => {
+  const chainId = useSelector(getCurrentChainId);
+  const [toastVisible, setToastVisible] = useState(false);
+  const t = useI18nContext();
+
+  const hideToast = useCallback(() => {
+    setToastVisible(false);
+  }, [setToastVisible]);
+
+  useEffect(() => {
+    let isMounted = true;
+    if (!confirmation) {
+      return undefined;
+    }
+    (async () => {
+      const lastInteractedConfirmationInfo =
+        await getLastInteractedConfirmationInfo();
+      const currentTimestamp = new Date().getTime();
+      if (
+        lastInteractedConfirmationInfo &&
+        lastInteractedConfirmationInfo.chainId !== chainId &&
+        currentTimestamp - lastInteractedConfirmationInfo.timestamp <=
+          MILLISECONDS_IN_ONE_MINUTE &&
+        isMounted
+      ) {
+        setToastVisible(true);
+        setTimeout(() => {
```

Comment:
Is it worth splitting this into a separate `TimeoutToast` component assuming that's going to be a common use case?

Reply:
I would prefer to create and plan separate task for this.